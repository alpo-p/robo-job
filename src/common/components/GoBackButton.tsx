import { useTheme } from '@react-navigation/native'
import React from 'react'
import IconButton from './IconButton'

interface P {
  onPress: () => void
  editStylesForChatScreen?: boolean
}

export default ({ onPress, editStylesForChatScreen }: P) => {
  const { colors } = useTheme()
  return (
    <IconButton
      iconName={
        editStylesForChatScreen ? 'chevron-back' : 'md-arrow-back-circle'
      }
      onPress={onPress}
      color={editStylesForChatScreen ? colors.text : colors.primary}
      style={{
        alignSelf: editStylesForChatScreen ? undefined : 'flex-start',
        marginLeft: 8,
        position: editStylesForChatScreen ? 'relative' : 'absolute',
        zIndex: 1,
        top: editStylesForChatScreen ? undefined : 48,
      }}
    />
  )
}
