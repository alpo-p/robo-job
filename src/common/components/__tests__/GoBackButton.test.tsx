import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react-native'
import GoBackButton from '../GoBackButton'

describe('GoBackButton', () => {
  const mockPress = jest.fn()

  it('should match snapshot', () => {
    const tree = renderer.create(<GoBackButton onPress={mockPress} />)
    expect(tree).toMatchSnapshot()
  })
  it('should register a press', () => {
    const { container } = render(<GoBackButton onPress={mockPress} />)
    fireEvent.press(container)
    expect(mockPress).toHaveBeenCalledTimes(1)
  })
})
