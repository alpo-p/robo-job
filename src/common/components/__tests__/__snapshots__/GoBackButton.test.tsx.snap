// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoBackButton should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "flex-start",
      "marginLeft": 8,
      "opacity": 1,
      "position": "absolute",
      "top": 48,
      "zIndex": 1,
    }
  }
>
  <Text />
</View>
`;
